src/presentation_exchange.rs

```diff
@@ -386,6 +493,109 @@ impl InputDescriptor {
+        }
+
+        let vp = &verifiable_presentation.0;
+
+        let vp_json: serde_json::Value =
+            from_value(vp.clone()).context("failed to parse value into json type")?;
+
+        if let Some(ConstraintsLimitDisclosure::Required) = self.constraints.limit_disclosure {
+            if self.constraints.fields().is_none() {
+                bail!("Required limit disclosure must have fields.")
+            }
+        };
+
+        if let Some(constraint_fields) = self.constraints.fields() {
+            for constraint_field in constraint_fields.iter() {
+                // Check if the filter exists if the predicate is present
+                // and set to required.
+                if let Some(Predicate::Required) = constraint_field.predicate() {
+                    if constraint_field.filter().is_none() {
+                        bail!("Required predicate must have a filter.")
+                    }
+                }
+
+                let mut selector = jsonpath_lib::selector(&vp_json);
+
+                // The root element is relative to the descriptor map path returned.
+                let Ok(root_element) = selector(descriptor_map.path()) else {
+                    bail!("Failed to select root element from verifiable presentation.")
+                };
+
+                let root_element = root_element
+                    .first()
+                    .ok_or(anyhow::anyhow!("Root element not found."))?;
+
+                let mut map_selector = jsonpath_lib::selector(root_element);
+
+                let validator = constraint_field.validator();
+
+                for field_path in constraint_field.path.iter() {
+                    let field_elements = map_selector(field_path)
+                        .context("Failed to select field elements from verifiable presentation.")?;
+
+                    // Check if the field matches are empty.
+                    if field_elements.is_empty() {
+                        if let Some(ConstraintsLimitDisclosure::Required) =
+                            self.constraints.limit_disclosure
+                        {
+                            bail!("Field elements are empty while limit disclosure is required.")
+                        }
+
+                        // According the specification, found here:
+                        // [https://identity.foundation/presentation-exchange/spec/v2.0.0/#input-evaluation](https://identity.foundation/presentation-exchange/spec/v2.0.0/#input-evaluation)
+                        // > If the result returned no JSONPath match, skip to the next path array element.
+                        continue;
+                    }
+
+                    // If a filter is available with a valid schema, handle the field validation.
+                    if let Some(Ok(schema_validator)) = validator.as_ref() {
+                        // TODO: possible trace a warning if a field is not valid.
```

Comment:
currently using `debug!` to trace filter validation errors. Possibly unnecessary.